numba/tests/test_runtests.py

```diff
@@ -120,20 +120,33 @@ def get_count(arg_list):
             self.assertEqual(total, included + excluded)
 
     def test_check_slice(self):
-        tmp = self.get_testsuite_listing(['-j','0,5,1'])
-        l = [x for x in tmp if x.startswith('numba.')]
-        self.assertEqual(len(l), 5)
+        tmpAll = self.get_testsuite_listing([])
+        tmp1 = self.get_testsuite_listing(['-j', '0:2'])
+        tmp2 = self.get_testsuite_listing(['-j', '1:2'])
+        lAll = {x for x in tmpAll if x.startswith('numba.')}
+        l1 = {x for x in tmp1 if x.startswith('numba.')}
+        l2 = {x for x in tmp2 if x.startswith('numba.')}
+        # The variance between two adjacent shards should be less than 5% of
+        # the total
+        self.assertLess(abs(len(l2) - len(l1)), len(lAll) / 20)
+        self.assertLess(len(l1), len(lAll))
+        self.assertLess(len(l2), len(lAll))
+        # Test should always run
+        self.assertIn("numba.tests.test_api.TestNumbaModule.test_numba_module",
+                      l1)
+        self.assertIn("numba.tests.test_api.TestNumbaModule.test_numba_module",
+                      l2)
 
     def test_check_slicing_equivalent(self):
         def filter_test(xs):
-            return [x for x in xs if x.startswith('numba.')]
+            return {x for x in xs if x.startswith('numba.')}
```

Comment:
Is the use of `set`s in this test potentially hiding bugs in that a set implicitly filters duplicates?

Reply:
There will be duplicates for any tests that are marked to run on all shards.

Reply:
btw, the duplicates will be the always running tests.